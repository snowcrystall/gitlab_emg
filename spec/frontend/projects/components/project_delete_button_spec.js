import { shallowMount } from '@vue/test-utils';
import ProjectDeleteButton from '~/projects/components/project_delete_button.vue';
import SharedDeleteButton from '~/projects/components/shared/delete_button.vue';

jest.mock('lodash/uniqueId', () => () => 'fakeUniqueId');

describe('Project remove modal', () => {
  let wrapper;

  const findSharedDeleteButton = () => wrapper.find(SharedDeleteButton);

  const defaultProps = {
    confirmPhrase: 'foo',
    formPath: 'some/path',
  };

  const createComponent = (props = {}) => {
    wrapper = shallowMount(ProjectDeleteButton, {
      propsData: {
        ...defaultProps,
        ...props,
      },
      stubs: {
        SharedDeleteButton,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('initialized', () => {
    beforeEach(() => {
      createComponent();
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('passes confirmPhrase and formPath props to the shared delete button', () => {
      expect(findSharedDeleteButton().props()).toEqual(defaultProps);
    });
  });
});
