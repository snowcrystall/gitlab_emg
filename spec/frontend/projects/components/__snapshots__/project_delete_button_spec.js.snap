// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project remove modal initialized matches the snapshot 1`] = `
<form
  action="some/path"
  method="post"
>
  <input
    name="_method"
    type="hidden"
    value="delete"
  />
   
  <input
    name="authenticity_token"
    type="hidden"
  />
   
  <gl-button-stub
    buttontextclasses=""
    category="primary"
    icon=""
    role="button"
    size="medium"
    tabindex="0"
    variant="danger"
  >
    Delete project
  </gl-button-stub>
   
  <gl-modal-stub
    actioncancel="[object Object]"
    actionprimary="[object Object]"
    dismisslabel="Close"
    footer-class="gl-bg-gray-10 gl-p-5"
    modalclass=""
    modalid="fakeUniqueId"
    ok-variant="danger"
    size="sm"
    title-class="gl-text-red-500"
    titletag="h4"
  >
     
    <div>
      <gl-alert-stub
        class="gl-mb-5"
        dismisslabel="Dismiss"
        primarybuttonlink=""
        primarybuttontext=""
        secondarybuttonlink=""
        secondarybuttontext=""
        title="You are about to permanently delete this project"
        variant="danger"
      >
        <gl-sprintf-stub
          message="Once a project is permanently deleted, it %{strongStart}cannot be recovered%{strongEnd}. Permanently deleting this project will %{strongStart}immediately delete%{strongEnd} its repositories and %{strongStart}all related resources%{strongEnd}, including issues, merge requests etc."
        />
      </gl-alert-stub>
       
      <p
        class="gl-mb-1"
      >
        Please type the following to confirm:
      </p>
       
      <p>
        <code
          class="gl-white-space-pre-wrap"
        >
          foo
        </code>
      </p>
       
      <gl-form-input-stub
        id="confirm_name_input"
        name="confirm_name_input"
        type="text"
      />
       
    </div>
  </gl-modal-stub>
</form>
`;
